pymc_marketing/mlflow.py

```diff
@@ -412,14 +460,319 @@ def log_inference_data(
     os.remove(save_file)
 
 
+def log_summary_metrics(
+    y_true: np.ndarray,
+    y_pred: np.ndarray,
+    metrics_to_calculate: list[str] | None = None,
+    hdi_prob: float = 0.94,
+) -> None:
+    """Log evaluation metrics produced by `pymc_marketing.mmm.evaluation.compute_summary_metrics()` to MLflow.
+
+    Parameters
+    ----------
+    y_true : np.ndarray
+        The true values of the target variable.
+    y_pred : np.ndarray
+        The predicted values of the target variable.
+    metrics_to_calculate : list of str or None, optional
+        List of metrics to calculate. If None, all available metrics will be calculated
+        see `evaluation.calculate_metric_distributions` for more info.
+    hdi_prob : float, optional
+        The probability mass of the highest density interval. Defaults to 0.94.
+
+    """
+    # Convert y_true and y_pred to numpy arrays if they're not already
+    y_true_np = y_true.to_numpy() if hasattr(y_true, "to_numpy") else np.array(y_true)
+    y_pred_np = y_pred.to_numpy() if hasattr(y_pred, "to_numpy") else np.array(y_pred)
+
+    metric_summaries = compute_summary_metrics(
+        y_true=y_true_np,
+        y_pred=y_pred_np,
+        metrics_to_calculate=metrics_to_calculate,
+        hdi_prob=hdi_prob,
+    )
+
+    for metric, stats in metric_summaries.items():
+        for stat, value in stats.items():
+            mlflow.log_metric(f"{metric}_{stat.replace('%', '')}", value)
```

Comment:
Same comment as above `log_metrics`

Reply:
Ahh this one is just because MLFlow doesn't support logging with "%" in the name - so it was either this or a dict comprehension followed by `log_metrics()` - which pretty much are same-same in my mind.

I added a comment though!